src/GameData.ts

```diff
@@ -156,9 +191,58 @@ export default class GameData {
         if (typeof result === "string") {
             message.channel.send(result);
             return;
+        } else {
+            message.channel.send(this.buildEmbed(result));
+        }
+    }
+
+    public buildEmbed(rawData : EmbeddableDatum) {
+        let embed = new Discord.RichEmbed()
+
+        switch(rawData.type) {
+            case "ChampionDatum":
+                embed.setThumbnail(`https://cdn.communitydragon.org/latest/champion/${rawData.id}/square.png`)
+                embed.setURL(this.sharedSettings.lookup.championUrl);
+                break;
+            case "ItemDatum":
+                var imageString = (`https://raw.communitydragon.org/latest/plugins${rawData.iconPath}`
+                    .replace("lol-game-data", "rcp-be-lol-game-data/global/default")
+                    .replace("/assets", "")
+                    .toLowerCase()
+                    );
+                embed.setThumbnail(imageString);
+                embed.setURL(this.sharedSettings.lookup.itemUrl);
+                break;
+            case "PerkDatum":
+                var imageString = (`https://raw.communitydragon.org/latest/plugins${rawData.iconPath}`
+                    .replace("lol-game-data", "rcp-be-lol-game-data/global/default")
+                    .replace("/assets", "")
+                    .toLowerCase()
+                    );
+                embed.setThumbnail(imageString);
+                embed.setURL(this.sharedSettings.lookup.perkUrl);
+                break;
+        }
+        delete rawData.type;
+        delete rawData.iconPath;
+        
+        embed.setTitle(rawData["name"]);
+
+        for (let [key, value] of Object.entries(rawData)) {
+            if (value != "") {
+                if (Array.isArray(value)) {
+                    if (value.length > 4) {
+                        embed.addField(key.toString().charAt(0).toUpperCase() + key.toString().slice(1), `${value.slice(0, 4).join(', ')} + ${value.length - 4} more...`)
+                    } else {
+                        embed.addField(key.toString().charAt(0).toUpperCase() + key.toString().slice(1), `${value.join(", ")}`)
+                    }
+                } else if (value.toString() != "") {
+                    embed.addField(key.toString().charAt(0).toUpperCase() + key.toString().slice(1), value, true);
```

Comment:
You can converting key to string all over the place. Would be easier to read to just convert once and store in a lcoal var.

`const keyStr = key.toString();`